log/levels/levels.go

```diff
@@ -29,8 +29,54 @@ type Levels struct {
 	warnValue  string
 	errorValue string
 	critValue  string
+
+	// FilterLogLevel is an option allowing logs below a certain level to be
+	// discarded rather than logged. For example, a consumer may set LogLevel
+	// to levels.Warn, and any logs to Debug or Info would become no-ops.
+	filterLogLevel LogLevel
+
+	// Promote errors
+	promoteErrors       bool
+	errorKey            string
+	promoteErrorToLevel LogLevel
+}
+
+// levelCommittedLogger embeds a log level that the user has selected and prepends
+// the associated level string as a prefix to every log line. Log level may be
+// affected by error promotion; and log lines may be suppressed based on the value
+// of FilterLogLevel.
+type levelCommittedLogger struct {
```

Comment:
With the default options for `filterLogLevel`, `promoteErrors`, `errorKey`, and `promoteErrorToLevel`, there shouldn't be any changes to the behavior of go-kit/log or go-kit/log/levels... Can you give a specific instance of existing behavior that would be altered?